cfme/infrastructure/provider/openstack_infra.py

```diff
@@ -106,3 +118,11 @@ def node_exist(self, name='my_node'):
         node_uuid = str(nodes_dict[name])
         for db_node in query.all():
             return db_node.hosts.name == str(node_uuid.uuid)
+
+    def load_all_provider_nodes(self):
```

Comment:
REQUIRED: I guess you've created navmazing destination - 'ProviderNodes' for replacing this method. Am I right ?

You could check whether destination exists in navmazing step and raise NavigationDestinationNotFound otherwise.